core/serialize/serialize.cc

```diff
@@ -317,6 +317,15 @@ void SerializerImpl::pickle(Pickler &p, Type *what) {
         p.putU4(alias->symbol._id);
     } else if (auto *lp = cast_type<LambdaParam>(what)) {
         p.putU4(8);
+        // special-case fixed bounds to preserve sharing when unpickling
+        if (lp->lower == lp->upper) {
+            p.putU1(1);
```

Comment:
I think we can pickle `nullptr` correctly, so this entire sections of `if`s can be simplified to

```
pickle(p, lp->lower.get());
pickle(p, lp->upper.get());
```

Reply:
I switched to what it's doing now because I was relying on `lower == upper` meaning that it was a fixed type member. Going back to relying on the flag on `definition` instead means that this special logic for detecting the fixed case is unnecessary.